modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/RasterLayerResponse.java

```diff
@@ -714,8 +683,15 @@ private MosaicOutput prepareResponse() throws DataSourceException {
                 }
             }
 
-            // prepare a blank response
-            return createBlankResponse();
+            // do we return a null (outside of the coverage) or a blank? The choice is "hard" as we
+            // might be in a hole of the coverage and not know it
+            if(!mosaicBBox.intersects((BoundingBox) ReferencedEnvelope.reference(coverageEnvelope)) &&
+                    !mosaicBBox.intersects((BoundingBox) ReferencedEnvelope.reference(rasterManager.spatialDomainManager.coverageBBox))) {
+                return null;
```

Comment:
Do you think it would make sense to add a log at FINE*** level reporting about no intersections at all?

Reply:
I'll add it